### PR TITLE
[System]: Enable some System.Net.Requests tests from CoreFX.

### PR DESCRIPTION
**Pull request by baulig**
### General Changes

- tests.make: add `$(XTEST_HARNESS_PATH)` to `TEST_MONO_PATH`, so we can find xunit.assert.

- corefx/SR.cs: add `System.Net.Requests` resources from corefx.

- HttpWebRequest: See below for detailed summary.

- Match parameter names of overridden base class members.

- Fixed order of some argument / state checks to match the tested behavior.

- Most async methods now throw exceptions instead of returning faulted tasks when called with incorrect arguments.

- HttpWebResponse.ContentType: use `string.empty` as default.

- WebConnectionStream.FlushAsync: override to return a finished task.

- HttpWebRequestTest.AllowReadStreamBuffering: removed this test.

- test-helpers/RemoteExecutorConsoleApp: silently ignore compiler-generated helper methods.

### Enabled tests from CoreFX

Added the following tests from external/corefx/src/System.Net.Requests/tests:

- AuthorizationTest, FileWebRequestTest, GlobalProxySelectionTest, HttpRequestCachePolicyTest, HttpWebRequestTest, HttpWebResponseHeaderTest, HttpWebResponseTest, LoggingTest, RequestStreamTest, WebRequestTest

Not working yet:

- AuthenticationManagerTest, FtpWebRequestTest

We are currently using
`../../../external/corefx/src/System.Net.Requests/tests/*.cs:AuthenticationManagerTest.cs,FtpWebRequestTest.cs`

### Summary of HttpWebRequest changes

- Host - we are now using the implementation from corefx.

- DefaultMaxResponseHeadersLength (unused) - fix default value.

- AllowReadStreamBuffering (unused) - allow this property to be read/written.

- ContinueTimeout - likewise.

- DefaultCachePolicy (unused) - allow this property to be read/written and set default value.

- DefaultMaximumErrorResponseLength (unused) - likewise.

- Connection - use implementation from corefx.

- Method - use implementation from corefx, which has additional error checks.

- MaximumResponseHeadersLength - use implementation from corefx, adding error checks.

- ReadWriteTimeout - likewise.

- ProtocolVersion - include parameter name in ArgumentException.

- TransferEncoding - use implementation from corefx.

- BeginGetRequestStream / GetRequestStreamAsync / GetRequestStream: throw exceptions when called with incorrect arguments rather than returning faulted tasks.

Contributes to #6987

### CODE REVIEW

File: mcs/class/System/System.Net/HttpWebRequest.cs [hunk 1]
```diff
@@ -393,23 +394,15 @@ void SetDateHeaderHelper (string headerName, DateTime dateTime)
 #if !MOBILE
 		[MonoTODO]
```
Comment (marek-safar) [inline mcs/class/System/System.Net/HttpWebRequest.cs]: Remove [MonoTODO]

File: mcs/class/System/System.Net/HttpWebRequest.cs [hunk 2]
```diff
@@ -393,23 +394,15 @@ void SetDateHeaderHelper (string headerName, DateTime dateTime)
...
-			}
-			set {
-				throw GetMustImplement ();
-			}
+			get { return defaultCachePolicy; }
+			set { defaultCachePolicy = value; }
 		}
 #endif
 
 		[MonoTODO]
```
Comment (marek-safar) [inline mcs/class/System/System.Net/HttpWebRequest.cs]: Remove [MonoTODO]

File: mcs/class/System/System.Net/HttpWebRequest.cs (hunk 1 above)
Comment (baulig) [reply mcs/class/System/System.Net/HttpWebRequest.cs]: We don't actually do anything with the value - shouldn't we keep the attribute until it's actually implemented?

File: mcs/build/tests.make [hunk 3]
```diff
@@ -262,7 +262,9 @@ ifdef HAVE_CS_XTESTS
 XTEST_HARNESS_PATH = $(topdir)/../external/xunit-binaries
 XTEST_HARNESS = $(XTEST_HARNESS_PATH)/xunit.console.exe
 XTEST_HARNESS_FLAGS := -noappdomain -noshadow -parallel none -nunit TestResult-$(PROFILE)-xunit.xml
-XTEST_TRAIT := -notrait category=failing -notrait category=nonmonotests -notrait Benchmark=true -notrait category=outerloop
+XTEST_TRAIT := -notrait category=failing -notrait category=nonmonotests -notrait category=nonnetfxtests -notrait Benchmark=true -notrait category=outerloop
```
Comment (marek-safar) [inline mcs/build/tests.make]: We cannot use `nonnetfxtests` that would disable tests we expect to pass. There should be only few cases where we need to disable such tests and should use Mono test moniker

**Comment by baulig** [comment]
We have some failing tests, which need to be investigated before we can merge this.

**Comment by baulig** [comment]
The failing tests have now been fixed, removing the do-not-merge tag.

**Comment by baulig** [comment]
@monojenkins commit api-diff

**Comment by baulig** [comment]
@monojenkins commit apidiff

**Comment by marek-safar** [comment]
@monojenkins commit apidiff

**Comment by baulig** [comment]
build

**Comment by baulig** [comment]
Created a new PR #7080, which is based on the current master and does not contain any merge conflicts.  We might want to close this one.

**Pull request closed without merging**